backend/dal/Helpers/Extensions/ProjectExtensions.cs

```diff
@@ -376,7 +377,210 @@ public static bool IsProjectInDraft(this Entity.Project project, ProjectOptions
         /// <returns></returns>
         public static bool IsProjectClosed(this Entity.Project project, ProjectOptions options)
         {
-            return options.ClosedStatus.Contains(project.Status.Code);
+            return options.ClosedStatus?.Contains(project.Status.Code) ?? throw new ConfigurationException("Project closed status have not been configured.");
+        }
+
+        /// <summary>
+        /// Merge the specified 'updatedProject' changes into the specified 'originalProject'.
+        /// </summary>
+        /// <param name="originalProject"></param>
+        /// <param name="updatedProject"></param>
+        /// <param name="context"></param>
+        /// <param name="options"></param>
+        public static void Merge(this Entity.Project originalProject, Entity.Project updatedProject, PimsContext context, ProjectOptions options)
+        {
+            // Update a project
+            context.Entry(originalProject).CurrentValues.SetValues(updatedProject);
+            context.SetOriginalRowVersion(originalProject);
+
+            // Update all properties
+            foreach (var property in updatedProject.Properties)
+            {
+                var existingProperty = originalProject.Properties.FirstOrDefault(b => b.PropertyType == Entity.PropertyTypes.Land
+                && b.ParcelId == property.ParcelId
+                && b.ProjectId == updatedProject.Id
+                ||
+                b.PropertyType == Entity.PropertyTypes.Building
+                && b.ProjectId == updatedProject.Id
+                && b.BuildingId == property.BuildingId);
+
+                if (existingProperty == null)
+                {
+                    //Todo: Navigation properties on project object were causing concurrency exceptions.
+                    var eproperty = property.PropertyType == Entity.PropertyTypes.Land ? context.Parcels.Find(property.ParcelId) : context.Buildings.Find(property.BuildingId) as Entity.Property;
+                    // Ignore properties that don't exist.
+                    if (eproperty != null)
+                    {
+                        if (property.PropertyType == Entity.PropertyTypes.Land)
+                        {
+                            var existingParcel = context.Parcels
```

Comment:
Actually, do we need to load the evaluations and fiscals here? it seems like we load those properties into existingParcel but then don't use them.

Reply:
Not sure I understand.  The update function will need to handle any changes made.  Additionally under most conditions it will need to recalculate the project financial values.